lib/beaker/dsl/helpers.rb

```diff
@@ -553,7 +553,7 @@ def with_puppet_running_on host, conf_opts, testdir = host.tmpdir(File.basename(
         end
 
         begin
-          backup_file = backup_the_file(host, host['puppetconfdir'], testdir, 'puppet.conf')
+          backup_file = backup_the_file(host, host.puppet['confdir'], testdir, 'puppet.conf')
```

Comment:
If 'with_puppet_running_on' can only be called by the master we should put in a check for that and fail out otherwise.